digitalocean/datasource_digitalocean_database_cluster.go

```diff
@@ -176,14 +176,20 @@ func dataSourceDigitalOceanDatabaseClusterRead(ctx context.Context, d *schema.Re
 				}
 			}
 
-			d.Set("host", db.Connection.Host)
-			d.Set("private_host", db.PrivateConnection.Host)
-			d.Set("port", db.Connection.Port)
-			d.Set("uri", db.Connection.URI)
-			d.Set("private_uri", db.PrivateConnection.URI)
-			d.Set("database", db.Connection.Database)
-			d.Set("user", db.Connection.User)
-			d.Set("password", db.Connection.Password)
+			if db.Connection != nil {
+				d.Set("host", db.Connection.Host)
+				d.Set("port", db.Connection.Port)
+				d.Set("uri", db.Connection.URI)
+				d.Set("database", db.Connection.Database)
+				d.Set("user", db.Connection.User)
+				d.Set("password", db.Connection.Password)
+			}
+
+			if db.PrivateConnection != nil {
+				d.Set("private_host", db.PrivateConnection.Host)
+				d.Set("private_uri", db.PrivateConnection.URI)
+			}
+
 			d.Set("urn", db.URN())
 			d.Set("private_network_uuid", db.PrivateNetworkUUID)
```

Comment:
You'll have to check for the existence of `db.PrivateNetworkUUID` as well, as `private_network_uuid` doesn't exist in the API response when the DB isn't assigned to a VPC at all (any DB created before VPCs were added).

Reply:
The type for a `db.PrivateNetworkUUID` is string. So if it does not exist, it be assigned the default value which is an empty string. The problem that causes the panic is that `db.PrivateConnection` is a struct. The default value of a struct is `nil`. So when you attempt to access an attribute (e.g. `db.PrivateConnection.URI`) it causes a panic as `nil` does not have any attributes.

Reply:
Ahh, OK, thanks for the explanation! I've not used Go yet :)